### ISSUE TITLE
arduino.js

### ISSUE THREAD

**Issue by S1SYPHOS** [stakx-io/highlight.php#1]
Hey there,
first of all, great work! One question though: Did you get `arduino` working?

Cheers!

**Comment by allejo** [comment]
If I remember correctly, the arduino language file didn't work because it extended another language and the conversion script didn't support extending languages, *I think*. I just haven't gotten around to trying to fix the issue and/or adding support for extending languages.

**Edit:**

Ah, it's not the PHP script, it's the `launcher.js` script that isn't importing other languages.

```
  var CPP = hljs.getLanguage('cpp').exports;
                                   ^

TypeError: Cannot read property 'exports' of undefined
    at module.exports (./highlight.php/tools/lib_highlight/build/lib/languages/arduino.js:2:36)
    at exportLang (export.js:37:11)
    at export.js:51:4
    at Array.forEach (<anonymous>)
    at export.js:50:9
    at FSReqWrap.oncomplete (fs.js:135:15)
```

**Comment by allejo** [comment]
The `launcher.js` script needs to be patched, but patching `tools/lib_highlight/build/lib/languages/arduino.js` manually works for now.

```js
var cppDef = require('./cpp.js');
hljs.registerLanguage('cpp', cppDef);

var CPP = hljs.getLanguage('cpp').exports;
```

**Comment by allejo** [comment]
Here's [the arduino language definition](https://github.com/stakx-io/highlight.php/blob/master/Highlight/languages/arduino.json) pushed to master.

The whole process should be fixed in e3519c01ea6a44c24bcfdfa1fd55f91f0f127aeb and no longer requires to be patched manually.
